validator-generator/src/main/java/io/avaje/validation/generator/ComponentReader.java

```diff
@@ -1,47 +1,95 @@
 package io.avaje.validation.generator;
 
-import java.util.Objects;
+import java.util.List;
+import java.util.Map;
 
 import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Modifier;
 import javax.lang.model.element.TypeElement;
 import javax.lang.model.type.TypeMirror;
 
 final class ComponentReader {
 
   private final ComponentMetaData componentMetaData;
+  private final Map<String, ComponentMetaData> privateMetaData;
 
-  ComponentReader(ComponentMetaData metaData) {
+  ComponentReader(ComponentMetaData metaData, Map<String, ComponentMetaData> privateMetaData) {
     this.componentMetaData = metaData;
+    this.privateMetaData = privateMetaData;
+  }
+
+  private static boolean isGeneratedComponent(TypeElement moduleType) {
+    return moduleType != null
+        && "io.avaje.validation.spi.GeneratedComponent"
+            .equals(moduleType.getSuperclass().toString());
   }
 
   void read() {
-    ProcessingContext.readExistingMetaInfServices().stream()
-      .map(APContext::typeElement)
-      .filter(Objects::nonNull)
-      .filter(t -> "io.avaje.validation.spi.GeneratedComponent".equals(t.getSuperclass().toString()))
-      .findFirst()
-      .ifPresent(moduleType -> {
-        componentMetaData.setFullName(moduleType.getQualifiedName().toString());
-        readMetaData(moduleType);
-      });
+    for (String fqn : ProcessingContext.readExistingMetaInfServices()) {
+      final TypeElement moduleType = APContext.typeElement(fqn);
+
+      if (isGeneratedComponent(moduleType)) {
+
+        if (hasPkgPrivate(moduleType)) {
+          componentMetaData.setFullName(fqn);
+          readMetaData(moduleType, componentMetaData);
```

Comment:
This looks like it should be "public" components going here ... and not the pkg private ones